tests/ssl-opt.sh

```diff
@@ -8815,6 +8836,18 @@ run_test    "export keys functionality" \
             -c "exported keylen is " \
             -c "exported ivlen is "
 
+# Test memory peak usage after handshake
+requires_config_enabled MBEDTLS_MEMORY_DEBUG
+requires_config_enabled MBEDTLS_MEMORY_BUFFER_ALLOC_C
+requires_config_value_at_most MBEDTLS_SSL_MAX_CONTENT_LEN 4096
+run_test    "Handshake memory usage peak" \
+            "$P_SRV debug_level=3 auth_mode=required force_version=tls1_2" \
+            "$P_CLI debug_level=3 force_version=tls1_2 \
```

Comment:
We need to use `max_frag_len` here, to test that with `max_frag_len` enabled, the server memory savings post-handshake vs without MFL post-handshake, is around 2*(16k-MFL), leaving some margin for robustness.

Reply:
Or to be more flexible we could set the expected savings to
`MBEDTLS_SSL_OUT_CONTENT_LEN` + `MBEDTLS_SSL_IN_CONTENT_LEN` -  2*MFL + margin

Reply:
I think it is god suggestion for future extension of this test, but now line "requires_config_value_at_most MBEDTLS_SSL_MAX_CONTENT_LEN 4096" is removed. Currently this test assume that content length is default